src/logic/calculate.js

```diff
@@ -0,0 +1,54 @@
+import operate from './operate';
+
+const calculate = (calcData, buttonName) => {
+  let { next, total, operation } = calcData;
+  const numbers = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
+
+  switch (buttonName) {
+    case '+/-':
+      total *= -1;
+      next *= -1;
+      break;
+    case 'AC':
+      total = null;
+      next = null;
+      break;
+    case '%':
+      operation = '%';
+      total = operate(total, next, operation).toString();
+      break;
+    case '.':
+      return 0;
```

Comment:
I think it should be `return 0.`

@helciodev update this.

kindly reply here, once you're done I will stay online for few minutes so that you won't need to send in another CRR.